go/integration_test/processor/processor_gas_test.go

```diff
@@ -0,0 +1,503 @@
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas + floria.TxDataZeroGasEIP2028*10 + excessGas/10,
+			},
+		},
+		"InputZerosInsufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + floria.TxDataZeroGasEIP2028*10 - 1,
+				Nonce:     4,
+				Input:     []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: 0,
+			},
+			OperaError: fmt.Errorf("gas too low"),
+		},
+		"InputNonZerosExact": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + floria.TxDataNonZeroGasEIP2028*10,
+				Nonce:     4,
+				Input:     []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas + floria.TxDataNonZeroGasEIP2028*10,
+			},
+		},
+		"InputNonZerosSufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + floria.TxDataNonZeroGasEIP2028*10 + excessGas,
+				Nonce:     4,
+				Input:     []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas + floria.TxDataNonZeroGasEIP2028*10 + excessGas/10,
+			},
+		},
+		"InputNonZerosInsufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + floria.TxDataNonZeroGasEIP2028*10 - 1,
+				Nonce:     4,
+				Input:     []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: 0,
```

Comment:
As argued above, I find it surprising that the reported `GasUsed` here is 0 instead of `GasLimit`. We need to check this.